L1Trigger/TrackTrigger/plugins/TTStubBuilder.cc

```diff
@@ -0,0 +1,327 @@
+  const TrackerGeometry* const theTrackerGeom = tGeomHandle.product();
+	
+  /// Prepare output
+  auto ttClusterDSVForOutput      = std::make_unique<edmNew::DetSetVector<TTCluster<Ref_Phase2TrackerDigi_>>>();
+  auto ttStubDSVForOutputTemp     = std::make_unique<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_>>>();
+  auto ttStubDSVForOutputAccepted = std::make_unique<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_>>>();
+  auto ttStubDSVForOutputRejected = std::make_unique<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_>>>();
+
+  /// Get the Clusters already stored away
+  edm::Handle< edmNew::DetSetVector< TTCluster< Ref_Phase2TrackerDigi_ > > > clusterHandle;
+  iEvent.getByToken( clustersToken, clusterHandle );
+
+  /// Get the maximum number of stubs per ROC
+  /// (CBC3-style)
+  //  unsigned maxStubs = theStackedTracker->getCBC3MaxStubs();
+  unsigned maxStubs = 3;
+
+  for (auto gd=theTrackerGeom->dets().begin(); gd != theTrackerGeom->dets().end(); gd++) {
+      DetId detid = (*gd)->geographicalId();
+      if(detid.subdetId()!=StripSubdetector::TOB && detid.subdetId()!=StripSubdetector::TID ) continue; // only run on OT
+      if(!tTopo->isLower(detid) ) continue; // loop on the stacks: choose the lower arbitrarily
+      DetId lowerDetid = detid;
+      DetId upperDetid = tTopo->partnerDetId(detid);
+      DetId stackDetid = tTopo->stack(detid);
+
+    /// Go on only if both detectors have Clusters
+    if ( clusterHandle->find( lowerDetid ) == clusterHandle->end() ||
+         clusterHandle->find( upperDetid ) == clusterHandle->end() )
+      continue;
+
+    /// Get the DetSets of the Clusters
+    edmNew::DetSet< TTCluster< Ref_Phase2TrackerDigi_ > > lowerClusters = (*clusterHandle)[ lowerDetid ];
+    edmNew::DetSet< TTCluster< Ref_Phase2TrackerDigi_ > > upperClusters = (*clusterHandle)[ upperDetid ];
+
+    /// If there are Clusters in both sensors
+    /// you can try and make a Stub
+    /// This is ~redundant
+    if ( lowerClusters.size() == 0 || upperClusters.size() == 0 )
+      continue;
+
+    /// Create the vectors of objects to be passed to the FastFillers
+    std::vector< TTCluster< Ref_Phase2TrackerDigi_ > > tempInner; 
+    std::vector< TTCluster< Ref_Phase2TrackerDigi_ > > tempOuter; 
+    std::vector< TTStub< Ref_Phase2TrackerDigi_ > >   tempAccepted; 
+    tempInner.clear();
+    tempOuter.clear();
+    tempAccepted.clear();
+
+    /// Get chip size information
+    const GeomDetUnit* det0 = theTrackerGeom->idToDetUnit( lowerDetid );
+    const PixelGeomDetUnit* pix0 = dynamic_cast< const PixelGeomDetUnit* >( det0 );
+    const PixelTopology* top0 = dynamic_cast< const PixelTopology* >( &(pix0->specificTopology()) );
+    const int chipSize = 2 * top0->rowsperroc(); /// Need to find ASIC size in half-strip units
+
+    std::map< int, std::vector< TTStub< Ref_Phase2TrackerDigi_ > > > moduleStubs; /// Temporary storage for stubs before max check
```

Comment:
@sviret @kpedro88 - stepping in for the L1 guys after some delay - not sure I get throw all of the comments now, but will start

Here - since you are searching this map every stub - pls use an unordered_map (though eventually I guess other solutions may need to be considered)